samples/sample_common/src/vaapi_utils_drm.cpp

```diff
@@ -22,169 +22,104 @@ or https://software.intel.com/en-us/media-client-solutions-support.
 #include "vaapi_utils_drm.h"
 #include "vaapi_allocator.h"
 #include <fcntl.h>
+#include <sys/ioctl.h>
 
-#include <dirent.h>
 #include <stdexcept>
+#include <vector>
 
+#include "vaapi_utils_drm.h"
 #include <drm_fourcc.h>
-#include <intel_bufmgr.h>
-
-#define MFX_PCI_DIR "/sys/bus/pci/devices"
-#define MFX_DRI_DIR "/dev/dri/"
-#define MFX_PCI_DISPLAY_CONTROLLER_CLASS 0x03
 
-struct mfx_disp_adapters
-{
-    mfxU32 vendor_id;
-    mfxU32 device_id;
+constexpr mfxU32 MFX_DRI_MAX_NODES_NUM = 16;
+constexpr mfxU32 MFX_DRI_RENDER_START_INDEX = 128;
+constexpr  mfxU32 MFX_DRM_DRIVER_NAME_LEN = 4;
+const char* MFX_DRM_INTEL_DRIVER_NAME = "i915";
+const char* MFX_DRI_NODE_RENDER_PATH = "/dev/dri/renderD";
+const char* MFX_DRI_NODE_CARD_PATH = "/dev/dri/card";
+
+enum {
+    MFX_DRI_NODE_RENDER,
+    MFX_DRI_NODE_CARD,
+    MFX_DRI_NODE_NUM
 };
 
-static int mfx_dir_filter(const struct dirent* dir_ent)
+bool get_drm_driver_name(int fd, char *name, int name_size)
 {
-    if (!dir_ent) return 0;
-    if (!strcmp(dir_ent->d_name, ".")) return 0;
-    if (!strcmp(dir_ent->d_name, "..")) return 0;
-    return 1;
-}
+    drm_version_t version = {};
+    version.name_len = name_size;
+    version.name = name;
 
-typedef int (*fsort)(const struct dirent**, const struct dirent**);
+    if (ioctl(fd, DRM_IOWR(0, drm_version), &version)) {
+        return false;
+    }
+    return true;
+}
 
-static mfxU32 mfx_init_adapters(struct mfx_disp_adapters** p_adapters)
+void get_intel_adapters(std::vector<mfxU32>& adapters)
 {
-    mfxU32 adapters_num = 0;
-    int i = 0;
-    struct mfx_disp_adapters* adapters = NULL;
-    struct dirent** dir_entries = NULL;
-    int entries_num = scandir(MFX_PCI_DIR, &dir_entries, mfx_dir_filter, (fsort)alphasort);
+    char driver_name[MFX_DRM_DRIVER_NAME_LEN + 1] = {};
 
-    char file_name[300] = {};
-    char str[16] = {0};
-    FILE* file = NULL;
-
-    for (i = 0; i < entries_num; ++i)
-    {
-        long int class_id = 0, vendor_id = 0, device_id = 0;
+    for (mfxU32 i = 0; i < MFX_DRI_MAX_NODES_NUM; ++i) {
+        std::string node = MFX_DRI_NODE_RENDER_PATH +
```

Comment:
So, we support only render nodes? Why we have cardN path then? or if you plan to support it where is a corresponding code?